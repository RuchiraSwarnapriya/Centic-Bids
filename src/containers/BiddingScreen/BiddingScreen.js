import React, { useState } from 'react';
import { View, Text, StyleSheet, SafeAreaView, TextInput, Alert } from 'react-native';
import Header from "../../components/appHeader/Header";
import { fetchItems } from '../../redux/actions/auctionItem';
import { updateItemDetails } from "../../services/auctionItems";
import { HOME } from '../../navigation/routePaths';
import { useDispatch, useSelector } from 'react-redux';
import CountDowner from "../../components/countDowner/CountDowner";
import ImageSlider from '../../components/imageSlider/ImageSlider';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { BID_TIME_OVER, BID_AMOUNT_EMPTY, BID_AMOUNT_GREATER, BID_CONFIRMATION, BID_PLACED } from "../../utils/alert-constants";
import { Colors } from "../../assets/colors";
import Button from "../../components/button/Button";


const BiddingScreen = ({ route, navigation }) => {

    const { title, description, basePrice, currentBid, expTime, images } = route.params;

    const dispatch = useDispatch();

    const [CurrentBid, setCurrentBid] = useState(currentBid);

    const [MyBid, setMyBid] = useState('');

    const fcmToken = useSelector(({ currentUser }) => currentUser.user.fcmToken);

    const bidderID = useSelector(({ currentUser }) => currentUser.user.id);

    const currentTime = new Date().getTime() / 1000;

    const remaningTime = expTime - currentTime;

    const bidOver = () => {
        alert(BID_TIME_OVER);
    };

    const verifyBid = () => {

        if (MyBid == 0 || null) {
            alert(BID_AMOUNT_EMPTY);
        }
        else if (MyBid <= currentBid) {
            alert(BID_AMOUNT_GREATER);
        }
        else {
            confirmBid();
        }

    };

    const confirmBid = () => {
        Alert.alert("Confirmation", BID_CONFIRMATION, [
            {
                text: "No",
                onPress: () => null,
                style: "cancel"
            },
            { text: "YES", onPress: () => placeBid() }
        ]);
    };

    const placeBid = () => {
        setCurrentBid(MyBid);
        console.log(MyBid)
        updateItemDetails("001", MyBid, fcmToken, bidderID);
        alert(BID_PLACED);
        dispatch(fetchItems());
        navigation.navigate(HOME);
    };

    return (
        <SafeAreaView style={styles.main}>
            <Header title={title} navigation={navigation} />
            <KeyboardAwareScrollView style={styles.container}>
                <View style={styles.carouselContainer}>
                    <ImageSlider images={images} imageStyle={styles.itemImage} />
                </View>
                <Text style={styles.description}>{description}</Text>
                <View style={styles.detailContainer}>
                    <View>
                        <Text style={styles.currentBid}>{basePrice}$</Text>
                        <Text style={[styles.placeHolder]}>Base Price</Text>
                    </View>
                    <View>
                        <Text style={styles.currentBid}>{CurrentBid}$</Text>
                        <Text style={styles.placeHolder}>Current Bid</Text>
                    </View>
                </View>
                <View style={styles.timerContainer}>
                    <CountDowner remaningTime={remaningTime} size={15} />
                    <Text style={[styles.placeHolder]}>Remaining Time</Text>
                </View>
                <TextInput style={styles.textInput} placeholder="Place your Bid here" keyboardType='numeric' onChangeText={value => setMyBid(value)} />
                {remaningTime <= 0 ?
                    <Button title="Place Bid" buttonStyle={styles.bidButton} onPress={() => bidOver()} />
                    :
                    <Button title="Place Bid" buttonStyle={styles.bidButton} onPress={() => verifyBid()} />
                }
            </KeyboardAwareScrollView>
        </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    main: {
        flex: 1,
        backgroundColor: Colors.white
    },
    container: {
        flex: 1,
        marginHorizontal: 16,
        marginBottom: 10
    },
    carouselContainer: {
        alignSelf: 'center',
        position: 'relative',
        top: 10,
        height: 200,
        width: 350,
    },
    itemImage: {
        alignItems: 'center',
        justifyContent: 'center',
        margin: 15,
        width: 330,
        height: 150,
        borderRadius: 5
    },
    description: {
        marginTop: 30,
        fontSize: 15,
        color: Colors.black,
        textAlign: 'center'
    },
    detailContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'baseline'
    },
    currentBid: {
        marginTop: 20,
        textAlign: 'center',
        fontSize: 30,
        fontWeight: 'bold'
    },
    placeHolder: {
        textAlign: 'center',
        fontSize: 12,
        fontWeight: 'bold',
        color: Colors.grey
    },
    timerContainer: {
        marginTop: -35,
        alignItems: 'center'
    },
    separator: {
        marginVertical: 8,
        borderBottomColor: '#FFFFFF',
        borderBottomWidth: StyleSheet.hairlineWidth,
    },
    textInput: {
        alignSelf: 'center',
        width: 300,
        textAlign: 'center',
        marginTop: 30,
        borderBottomColor: Colors.grey,
        borderBottomWidth: 1,
        fontSize: 25,
        fontWeight: 'bold'
    },
    bidButton: {
        width: 300,
        height: 40,
        backgroundColor: Colors.green,
        alignSelf: 'center',
        marginTop: 40,
        justifyContent: 'center',
        alignItems: 'center',
    },
});

export default BiddingScreen
