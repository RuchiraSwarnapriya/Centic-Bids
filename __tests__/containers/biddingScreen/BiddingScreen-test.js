import React from 'react';
import renderer from 'react-test-renderer';
import BiddingScreen from "../../../src/containers/biddingScreen/BiddingScreen";


jest.mock('@react-native-firebase/auth');

jest.mock("react-native/Libraries/EventEmitter/NativeEventEmitter");

jest.mock("react-redux");

jest.useFakeTimers();



describe('init', () => {
    test('renders correctly', () => {
        const images = [];
        const mockedParams = {
            route: { params: { titile: 'test-title', description: 'test-desc', basePrice: 'test-basePrice', currentBid: 'whatever-id', expTime: 'test-expTime', images: images  } },
            navigation: ''
        };

        const tree = renderer.create(<BiddingScreen {...mockedParams} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});