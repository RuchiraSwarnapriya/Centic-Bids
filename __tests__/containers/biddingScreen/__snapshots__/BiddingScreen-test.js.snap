// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 10,
        "marginRight": 15,
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 25,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïˆˆ
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
        "marginBottom": 10,
        "marginHorizontal": 16,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 200,
            "position": "relative",
            "top": 10,
            "width": 350,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/images/no_preview.png",
                }
              }
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "height": 120,
                    "justifyContent": "center",
                    "margin": 15,
                    "width": 120,
                  },
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "height": 150,
                    "justifyContent": "center",
                    "margin": 15,
                    "width": 330,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 15,
            "marginTop": 30,
            "textAlign": "center",
          }
        }
      >
        test-desc
      </Text>
      <View
        style={
          Object {
            "alignItems": "baseline",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginTop": 20,
                "textAlign": "center",
              }
            }
          >
            test-basePrice
            $
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#808080",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            Base Price
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginTop": 20,
                "textAlign": "center",
              }
            }
          >
            whatever-id
            $
          </Text>
          <Text
            style={
              Object {
                "color": "#808080",
                "fontSize": 12,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Current Bid
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": -35,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 5,
                        "justifyContent": "center",
                        "marginHorizontal": 2,
                      },
                      Object {
                        "height": 39,
                        "width": 34.5,
                      },
                      Object {
                        "backgroundColor": "#FAB913",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontVariant": Array [
                            "tabular-nums",
                          ],
                          "fontWeight": "bold",
                        },
                        Object {
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#000",
                        },
                      ]
                    }
                  >
                    -NaN
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "white",
                      "marginVertical": 2,
                    },
                    Object {
                      "fontSize": 8.333333333333334,
                    },
                    Object {
                      "color": "#000",
                    },
                  ]
                }
              >
                Days
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 5,
                        "justifyContent": "center",
                        "marginHorizontal": 2,
                      },
                      Object {
                        "height": 39,
                        "width": 34.5,
                      },
                      Object {
                        "backgroundColor": "#FAB913",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontVariant": Array [
                            "tabular-nums",
                          ],
                          "fontWeight": "bold",
                        },
                        Object {
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#000",
                        },
                      ]
                    }
                  >
                    -NaN
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "white",
                      "marginVertical": 2,
                    },
                    Object {
                      "fontSize": 8.333333333333334,
                    },
                    Object {
                      "color": "#000",
                    },
                  ]
                }
              >
                Hours
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 5,
                        "justifyContent": "center",
                        "marginHorizontal": 2,
                      },
                      Object {
                        "height": 39,
                        "width": 34.5,
                      },
                      Object {
                        "backgroundColor": "#FAB913",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontVariant": Array [
                            "tabular-nums",
                          ],
                          "fontWeight": "bold",
                        },
                        Object {
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#000",
                        },
                      ]
                    }
                  >
                    -NaN
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "white",
                      "marginVertical": 2,
                    },
                    Object {
                      "fontSize": 8.333333333333334,
                    },
                    Object {
                      "color": "#000",
                    },
                  ]
                }
              >
                Minutes
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 5,
                        "justifyContent": "center",
                        "marginHorizontal": 2,
                      },
                      Object {
                        "height": 39,
                        "width": 34.5,
                      },
                      Object {
                        "backgroundColor": "#FAB913",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontVariant": Array [
                            "tabular-nums",
                          ],
                          "fontWeight": "bold",
                        },
                        Object {
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#000",
                        },
                      ]
                    }
                  >
                    -NaN
                  </Text>
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "white",
                      "marginVertical": 2,
                    },
                    Object {
                      "fontSize": 8.333333333333334,
                    },
                    Object {
                      "color": "#000",
                    },
                  ]
                }
              >
                Seconds
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#808080",
                "fontSize": 12,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          Remaining Time
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Place your Bid here"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "#808080",
            "borderBottomWidth": 1,
            "fontSize": 25,
            "fontWeight": "bold",
            "marginTop": 30,
            "textAlign": "center",
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#008000",
            "borderRadius": 20,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 40,
            "opacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          Place Bid
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
